src/PowerShellEditorServices.Host/CodeLens/PesterCodeLensProvider.cs

```diff
@@ -63,12 +74,43 @@ public PesterCodeLensProvider(EditorSession editorSession)
                     new ClientCommand(
                         "PowerShell.RunPesterTests",
                         "Debug tests",
-                        new object[] { scriptFile.ClientFilePath, true /* Run in debugger */, pesterSymbol.TestName })),
+                        new object[] { scriptFile.ClientFilePath, true /* Run in debugger */, pesterSymbol.TestName, describeBlockLineNumber })),
             };
 
             return codeLensResults;
         }
 
+        /// <summary>
+        /// Used to determine the value of <see cref="_pesterV4_6_0_OrHigherAvailable"/> as a background task.
+        /// </summary>
+        private bool DeterminePesterVersion()
+        {
+            var powerShell = new PSCommand();
+            powerShell.AddCommand("Get-Module")
+                      .AddParameter("ListAvailable")
+                      .AddParameter("Name", "Pester");
+
+            IEnumerable<PSObject> result = Task.Run(() =>
+            {
+                return _editorSession.PowerShellContext.ExecuteCommandAsync<PSObject>(powerShell);
+            }).Result;
+
+            if (result != null && result.Any())
```

Comment:
Style suggestion to prevent indentation running away (["if you need more than 3 levels, you're screwed"](https://www.kernel.org/doc/html/v4.10/process/coding-style.html)):

```csharp
if (result == null) // result.Any() is redundant, since foreach is below
{
    return false;
}

foreach (PSObject module in result)
{
    if (module.BaseObject is PSModuleInfo psModuleInfo
        && psModuleInfo.Version >= s_pesterLineInvocationVersion)
    {
        return true;
    }
}

return false;
```